File: codegen/chains-go/asset.go
package chains

type Asset interface {
	ID() string
	Name() string
	// Address is for token
	Address() string
	Description() string
	Symbol() string
	Decimals() uint
	Type() string
	AssetType() AssetType
	BIP44CoinType() int
	Website() string
	Explorer() string
	// Icon return raw svg as string
	Icon() string
}

File: codegen/chains-rs/src/asset.rs
use crate::types::AssetType;

pub trait Asset {
    fn id(&self) -> &str;
    fn name(&self) -> &str;
    fn address(&self) -> &str;
    fn description(&self) -> &str;
    fn symbol(&self) -> &str;
    fn decimals(&self) -> u32;
    fn asset_type(&self) -> AssetType;
    fn asset_kind(&self) -> &str;
    fn bip44_coin_type(&self) -> isize;
    fn website(&self) -> &str;
    fn explorer(&self) -> &str;
    fn icon(&self) -> &str;
}
